natlib/svcd.c

```diff
@@ -185,3 +188,82 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+
+//////////////////////////////////////////////////////////////////////////////
+// SVCD.subdispatch() implementation
+// Maintainer: Running with scissors
+//////////////////////////////////////////////////////////////////////////////
+
+static int svcd_subdispatch(lua_State *L) {
+  //local parr = storm.array.fromstr(pay)
+  size_t len;
+  char *parr = lua_tolstring(L, 1, &len);
```

Comment:
`char` is not guaranteed to be an 8 bit type, you should use uint8_t instead (because arr.fromstr returns an 8 bit array)